spec/plugins/view_component_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'jekyll'
+require 'liquid'
+require_relative '../../_plugins/view_component.rb'
+
+RSpec.describe ComponentTag do
+  subject(:rendered) do
+    converter.convert(Liquid::Template.parse(content).render!(payload, { registers: { site: } }))
+  end
+  let(:payload) { {} }
+  let(:content) { '' }
+  let(:site) { Jekyll::Site.new(Jekyll::Configuration.from({})) }
+  let(:converter) { site.converters.find { |c| c.instance_of?(Jekyll::Converters::Markdown) } }
+  let(:test_component) { Class.new(BaseComponent) }
+
+  before do
+    stub_const('TestComponent', test_component)
+  end
+
+  context 'with passthrough content' do
+    let(:content) { "{% component test %}con\ntent{% endcomponent %}" }
+    let(:test_component) do
+      Class.new(BaseComponent) do
+        def call
+          content
+        end
+      end
+    end
+
+    it 'renders markdownified content as a single line' do
+      expect(rendered).to eq("<p>con tent</p>\n")
+    end
+  end
+
+  context 'with component rendered html content' do
+    let(:content) { '{% component test %}content{% endcomponent %}' }
+    let(:test_component) do
+      Class.new(BaseComponent) do
+        def call
+          tag.div(content)
+        end
+      end
+    end
+
+    it 'renders html without markdownification' do
+      expect(rendered).to eq("<div>content</div>\n")
+    end
+  end
+
+  context 'with string param' do
+    let(:content) { '{% component test arg="string" %}content{% endcomponent %}' }
+    let(:test_component) do
+      Class.new(BaseComponent) do
+        def initialize(arg:)
+          @arg = arg
+        end
+
+        def call
+          tag.div(@arg.class.name + @arg.to_s)
```

Comment:
WDYT of making the test content like this?
```suggestion
          tag.div("arg_class=#{@arg.class.name} arg=#{@arg}")
```
then the string to match would be like:
```
<div>arg_class=String arg=string</div>
```
might be a bit easier to read?

Reply:
>WDYT of making the test content like this?

Sounds good to me! Updated in 33b67a6.